packages/clay-form/src/MultiSelect.tsx

```diff
@@ -0,0 +1,283 @@
+	const [isFocused, setIsFocused] = useState();
+
+	useLayoutEffect(() => {
+		if (sourceItems) {
+			const matchedItems = sourceItems.filter(
+				item => inputValue && item.match(inputValue)
+			);
+
+			setActive(matchedItems.length !== 0);
+		}
+	}, [sourceItems, inputValue]);
+
+	const setNewValue = (newVal: string) => {
+		onItemsChange([...items, newVal]);
+
+		onInputChange('');
+	};
+
+	const handleKeyDown = (event: React.KeyboardEvent<HTMLInputElement>) => {
+		const {keyCode} = event;
+
+		if (inputValue && DELIMITER_KEYS.includes(keyCode)) {
+			event.preventDefault();
+
+			setNewValue(inputValue);
+		} else if (
+			!inputValue &&
+			keyCode === BACKSPACE_KEY &&
+			inputRef.current &&
+			lastItemRef.current
+		) {
+			inputRef.current.blur();
+			lastItemRef.current.focus();
+		}
+	};
+
+	const handlePaste = (event: React.ClipboardEvent<HTMLInputElement>) => {
+		const pastedText = event.clipboardData.getData('Text');
+
+		const pastedItems = pastedText
+			.split(',')
+			.map(item => item.trim())
+			.filter(Boolean);
+
+		if (pastedItems.length > 0) {
+			event.preventDefault();
+
+			onItemsChange([...items, ...pastedItems]);
+		}
+	};
+
+	return (
+		<div className="form-group" ref={forwardRef}>
+			{label && <label>{label}</label>}
+
+			<div className="input-group input-group-stacked-sm-down">
+				<div
+					className={classNames('input-group-item', {
+						focus: isFocused && isValid,
+						['has-error']: !isValid,
+					})}
+				>
+					<div className="form-control form-control-tag-group">
+						{items.map((item, i) => {
+							const removeItem = () =>
+								onItemsChange([
+									...items.slice(0, i),
+									...items.slice(i + 1),
+								]);
+
+							return (
+								<ClayLabel
+									closeButtonProps={{
+										onClick: removeItem,
+									}}
+									key={i}
+									onKeyDown={({keyCode}) => {
+										if (keyCode !== BACKSPACE_KEY) {
+											return;
+										}
+
+										if (inputRef.current) {
+											inputRef.current.focus();
+										}
+
+										removeItem();
+									}}
+									ref={
+										i === items.length - 1
+											? lastItemRef
+											: undefined
+									}
+									spritemap={spritemap}
+									tabIndex={0}
+								>
+									{item}
+								</ClayLabel>
+							);
+						})}
+
+						<input
+							{...otherProps}
+							className="form-control-inset"
+							onBlur={() => setIsFocused(false)}
+							onChange={event =>
+								onInputChange(
+									event.target.value.replace(',', '')
+								)
+							}
+							onFocus={() => setIsFocused(true)}
+							onKeyDown={handleKeyDown}
+							onPaste={handlePaste}
+							ref={inputRef}
+							type="text"
+							value={inputValue}
+						/>
+
+						{sourceItems && (
+							<ClayAutocomplete.DropDown
+								active={active}
+								alignElementRef={forwardRef}
+								onSetActive={setActive}
+							>
+								<ClayDropDown.ItemList>
+									{sourceItems
+										.filter(
```

Comment:
We have to take some care with our filter here, especially for use cases of the Portal where the filter is made in the backend and data has already been filtered.

Reply:
The backend can still provide a pre-filtered list. Since we specifically require an array of strings for `sourceItems`, it should be fine that we filter it based on string matching the `inputValue`. If this becomes a problem in the future we can add an additional API. I'd rather start simple at first and expand where needed.

Reply:
What I mean is, the backend gets the value of the input and filters with that value so the data is already filtered with the value of the input and if we do the filter here again it will just be something unnecessary. We already had the use case for Documents and Media.